src/components/SectionRow.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import {IInfographicsRow, IInfographicsSection} from '../types/IInfographicsSection'
+import InfographicsSection from './InfographicsSection'
+
+interface IProps {
+    row: IInfographicsRow,
+    selectedSection: IInfographicsSection | null,
+    handleToggleSelectSection: (section: IInfographicsSection) => void,
+    handleDeleteSection: (section: IInfographicsSection) => void,
+}
+
+const SectionRow = (props: IProps) => {
+    const {row, selectedSection, handleDeleteSection, handleToggleSelectSection} = props
+    const {sections, height} = row
+
+    const generateColumnTemplate = (): string => {
+        let template = ''
+        row.sections.forEach(section => {
+            template += `${section.widthPercent}% `
+        })
+
+        return template.trim()
+    }
+
+    const style = {
+        height: height,
+        display: 'grid',
+        gridTemplateColumns: generateColumnTemplate(),
+    }
+
+    return (
+        <div className={'section-row'} style={style}>
+            {sections.map(section => {
+                let isActive = false
+                if (selectedSection) {
+                    isActive = section.id === selectedSection.id
+                }
+                return <InfographicsSection
+                    key={section.id}
+                    section={{...section, isActive}}
```

Comment:
mej na pameti, ze kdyz to spojujes primo v returnu, tak se ti to pri re-renderu spojuje znovu a znovu, je dobre to mit ulozene v konstante